dvc/repo/experiments/remove.py

```diff
@@ -61,3 +69,17 @@ def _get_ref(ref_infos, name, cur_rev):
         msg.extend([f"\t{info}" for info in ref_infos])
         raise InvalidArgumentError("\n".join(msg))
     return ref_infos[0]
+
+
+def _remove_exp_by_ref_or_rev(repo, ref_or_rev: str):
+    ref_info = _get_exp_ref(repo, ref_or_rev)
+    if ref_info is not None:
+        remove_exp_refs(repo.scm, [ref_info])
+    else:
+        stash_index = _get_exp_stash_index(repo, ref_or_rev)
+        if stash_index is None:
+            raise InvalidArgumentError(
+                f"'{ref_or_rev}' is neither a valid experiment reference"
+                " nor a queued experiment revision"
```

Comment:
Quite weird here, because the asymmetric in committed and queued experiments.

Reply:
I would separate this into two methods, one for removing a complete experiment ref (only by ref name) and one for removing stash entries (by either name or revision).

Reply:
But without a flag, we can not know if the input is a ref or a revision. Some ref might use a shortcut name for example.
`epoch_1_layers3` - > `ep1la3` hard to distinguish them.

Reply:
Right, you would start by calling
`_remove_exp_ref(name)`.

If that fails because an exact exp ref match for `name` does not exist, then you would try the stash remove function. Which also needs to try for an exact name match first, and then handling it as git rev second.

Reply:
Basically you can parse the list of names that we get at the start into two separate lists before trying to remove anything at all. One list that matches exact refs, and a second list that does not match exact refs. Everything in the second list is either a stash entry (by name or rev) or is invalid entirely.